docs/enhancements/0001_project_enhancements.md

```diff
@@ -0,0 +1,103 @@
+# 0001: Project process refinement for 2023
+
+## Summary
+
+As our project grows we need to scale our collaboration processes to meet the needs of a growing community. The 
+approchability of a project is important to foster continued growth and ensure the mission and, perhaps just 
+as important, non-goals of the project are clearly understood. This enhancement proposes some next steps to take 
+in the refinement of kcp's existing processes.
+
+## Motivation
+
+Over the past year there has been a drive to show the "realness" of the kcp project. That investment has been 
+made by contributing enough code to each of the core features of kcp that an early adopter could, with enough drive,
+integrate with the APIs to understand the value proposition. Optimizing for that goal has allowed quick progress but
+at the expense of things like documentation.
+
+Within kcp there are also a few logical projects that can be separated in order to provide more clarity on the 
+direction of each project and facilitate growing contributors in those areas.
+
+In order to keep growing it is essential that the kcp project provides an easily approachable method for consumers to
+quickly understand the goals of the project, see the current state of features, and run the latest build. It is also
+important that potential contributors be able to quickly setup their development environment, understand the 
+engagement model for development, and be able to discover the right forums for their area of interest.
+
+This document proposes that now is the right time to take these steps in order to facilitate objectives in 2023.
+
+### Goals
+
+1. Agree on the need for an official enhancement process
+2. Agree on the need to split the workloads and control plane areas of interest
+3. Agree on the need to create a formal definition of done that includes documentation and testing requirements
+4. Agree on the need to identify graduation criteria of existing APIs
+5. Assign ownership of the above for definition and implementation
+
+### Non-Goals
+
+1. Actually define the specifics of each of the goals - this should be up to the community
+
+## Proposal
+
+### Implement an enhancement proposal process
```

Comment:
Not every logical project may have reached the same maturity level. Would it make sense to have the enhancement process per project or at least in what extend it gets consumed ? Lighter for younger ones and more diligent for more mature ones.

Reply:
Part of me feels like the process itself is likely the same: plan your work in a written, discoverable fashion then work your plan. What may be the delineating factor is the content.

For exploratory work I'd probably expect a series of smaller, learning-driven steps to a proposed design. More of the "scientific method" approach of iterating on a hypothesis but still under the guideline of answering a specific question or meeting a specific goal that can be laid out clearly. Maybe there is even a state that the proposal stays in while certain areas are proven.  Probably could use some advice here on how that may work in other communities.

For something a bit more stable, say introducing an API or a refactoring, there is probably much more certainty in a clearly outlined path.

Thoughts on that explanation of different content vs different process?

Reply:
Let's start with a single enhancements repo (similar to Kubernetes) and we can subdivide that repo as needed.

Reply:
Also we can be nimble with our processes per project.